src/provider/provider_os_memory_linux.c

```diff
@@ -117,22 +117,39 @@ long os_get_mempolicy(int *mode, unsigned long *nodemask, unsigned long maxnode,
                          MPOL_F_NODE | MPOL_F_ADDR);
 }
 
-int os_mmap(void *addr, size_t length, int prot, int flags, int fd, long offset,
-            void **out_addr) {
-    if (out_addr == NULL) {
-        assert(0);
-        return -1;
-    }
+int os_mmap_aligned(void *hint_addr, size_t length, size_t alignment, int prot,
+                    int flags, int fd, long offset, void **out_addr) {
+    assert(out_addr);
+
+    size_t extended_length = length + alignment;
```

Comment:
if the passed `alignment` is less than the page size the `mmap` guarantees that the allocated range is page aligned. But according to the current logic `head_len` will be `0`, but `tail_len` will be equal to `alignment` and `munmap` will be called.
Should we optimize this case or it's not worth it?

Reply:
Fixed.
Done.

Reply:
Is it OK now?